### PR TITLE
Janitors(and jani ert) dont miss disposal shots anymore

### PR DESCRIPTION
**Pull request by 1080pCat**
## What Does This PR Do

If you're a (ert)janitor you wont miss landing disposal shots anymore

Also adds the comment "Lets bartender and chef mount tables faster" to the table leap trait, as it was missing it
## Why It's Good For The Game

Janitors have been doing their job for a long time, you really think they'd be missing their shots at this point? Its mostly a QOL thing though.

The added comment is for standardization 
## Testing

compiled, lots of throwing stuff into trashbins
## Changelog
:cl:
tweak: (ERT)janitors no longer miss their disposal shots.
/:cl:

### CODE REVIEW

File: code/__HELPERS/trait_helpers.dm [hunk 1]
```diff
@@ -224,6 +224,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_MAGPULSE "magnetificent" // Used for anything that is magboot related
 #define TRAIT_NOSLIP "noslip"
 
+
```
Comment (DGamerL) [inline code/__HELPERS/trait_helpers.dm:227]:
```suggestion
```

**Comment by matttheficus** [comment]
Should probably just be a trait applied on job only. Having it attached to the HUDs is kinda ehhhhhhhh.

File: code/modules/clothing/glasses/hudglasses.dm [hunk 2]
```diff
@@ -251,12 +251,31 @@
...
+	if(slot == SLOT_HUD_GLASSES)
+		ADD_TRAIT(H, TRAIT_NEVER_MISSES_DISPOSALS, CLOTHING_TRAIT)
+
+/obj/item/clothing/glasses/hud/janitor/sunglasses/dropped(mob/living/carbon/human/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.get_item_by_slot(SLOT_HUD_GLASSES) == src)
+		REMOVE_TRAIT (H, TRAIT_NEVER_MISSES_DISPOSALS, CLOTHING_TRAIT)
```
Comment (DGamerL) [inline code/modules/clothing/glasses/hudglasses.dm:277]:
```suggestion
		REMOVE_TRAIT(H, TRAIT_NEVER_MISSES_DISPOSALS, CLOTHING_TRAIT)
```

File: code/modules/recycling/disposal.dm [hunk 3]
```diff
@@ -527,7 +527,7 @@
 		var/obj/item/I = mover
 		if(istype(I, /obj/item/projectile))
 			return
-		if(prob(75) || (istype(mover.throwing.thrower) && HAS_TRAIT(mover.throwing.thrower, TRAIT_BADASS)))
+		if(prob(75) || (istype(mover.throwing.thrower) && (HAS_TRAIT(mover.throwing.thrower, TRAIT_BADASS) || HAS_TRAIT(mover.throwing.thrower, TRAIT_NEVER_MISSES_DISPOSALS))))
 			I.forceMove(src)
 			for(var/mob/M in viewers(src))
 				M.show_message("\the [I] lands in \the [src].", 3)
```
Comment (DGamerL) [inline code/modules/recycling/disposal.dm:533]:
```suggestion
				M.show_message("[I] lands in [src].", 3)
```

File: code/modules/recycling/disposal.dm [hunk 4]
```diff
@@ -527,7 +527,7 @@
 		var/obj/item/I = mover
 		if(istype(I, /obj/item/projectile))
 			return
-		if(prob(75) || (istype(mover.throwing.thrower) && HAS_TRAIT(mover.throwing.thrower, TRAIT_BADASS)))
+		if(prob(75) || (istype(mover.throwing.thrower) && (HAS_TRAIT(mover.throwing.thrower, TRAIT_BADASS) || HAS_TRAIT(mover.throwing.thrower, TRAIT_NEVER_MISSES_DISPOSALS))))
```
Comment (DGamerL) [inline code/modules/recycling/disposal.dm:530]: Not sure about this line, it is getting quite convoluted, it might be time that this proc gets refactored?

**Comment by SynthTwo** [comment]
I agree making it a job trait only to be honest. It's not a bad idea, it's just weird if they are wearing a janihud. Makes a ton of sense for Janitor ERTs too

File: code/__HELPERS/trait_helpers.dm [hunk 5]
```diff
@@ -226,13 +226,15 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
...
 #define TRAIT_HOLY "is_holy" // The mob is holy in regards to religion
+#define TRAIT_NEVER_MISSES_DISPOSALS "trait_never_misses_disposals" // For janitors landing disposal throws
 
 //***** ITEM AND MOB TRAITS *****//
 /// Show what machine/door wires do when held.
 #define TRAIT_SHOW_WIRE_INFO "show_wire_info"
 ///Immune to the SM / makes you immune to it when worn
 #define TRAIT_SUPERMATTER_IMMUNE "supermatter_immune"
 
+
```
Comment (DGamerL) [inline code/__HELPERS/trait_helpers.dm:237]:
```suggestion
```

File: code/_globalvars/traits.dm [hunk 6]
```diff
@@ -87,7 +87,8 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_BADASS" = TRAIT_BADASS,
 		"TRAIT_FORCED_STANDING" = TRAIT_FORCED_STANDING,
 		"TRAIT_NOSLIP" = TRAIT_NOSLIP,
-		"TRAIT_MAGPULSE" = TRAIT_MAGPULSE
+		"TRAIT_MAGPULSE" = TRAIT_MAGPULSE,
+		"TRAIT_NEVER_MISSES_DISPOSALS" = TRAIT_NEVER_MISSES_DISPOSALS
```
Comment (DGamerL) [inline code/_globalvars/traits.dm:91]: Move this to the mind trait section

**Review by DGamerL — CHANGES_REQUESTED** [review]
You have to do a merge master, there is currently a stealth conflict due to more mind traits that have been added in the meantime

File: code/modules/recycling/disposal.dm (hunk 4 above)
Comment (lewcc) [reply code/modules/recycling/disposal.dm:531]: Probably fine for now.

**Review by lewcc — APPROVED** [review]
LGTM

**Pull request merged**